pkg/internal/controller/controller.go

```diff
@@ -321,6 +322,15 @@ func (c *Controller) reconcileHandler(ctx context.Context, obj interface{}) {
 	}
 }
 
+func (c *Controller) doReconcile(ctx context.Context, req reconcile.Request) (_ reconcile.Result, err error) {
+	defer func() {
```

Comment:
This should be optional and opt-in, similiar to what upstream does: https://github.com/kubernetes/apimachinery/blob/f916759cb6b8547418dc7708876ecab5c1961448/pkg/util/runtime/runtime.go#L29

Also, please add tests for this.

Reply:
Okay. I have added tests and changed all `recover()` to `utilruntime.HandleCrash()`, so that developers can set it via `utilruntime. ReallyCrash`.